server/store/datastore/cron.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastore
+
+import (
+	"fmt"
+
+	"github.com/woodpecker-ci/woodpecker/server/model"
+
+	"xorm.io/builder"
+)
+
+func (s storage) CronCreate(job *model.Cron) error {
+	if job.RepoID == 0 || job.Name == "" {
+		return fmt.Errorf("repoID and Name required")
+	}
+	_, err := s.engine.Insert(job)
+	return err
+}
+
+func (s storage) CronFind(repo *model.Repo, id int64) (*model.Cron, error) {
+	cronJob := &model.Cron{
+		RepoID: repo.ID,
+		ID:     id,
+	}
+	return cronJob, wrapGet(s.engine.Get(cronJob))
+}
+
+func (s storage) CronList(repo *model.Repo) ([]*model.Cron, error) {
+	cronJobs := make([]*model.Cron, 0, perPage)
+	return cronJobs, s.engine.Where("repo_id = ?", repo.ID).Find(&cronJobs)
+}
+
+func (s storage) CronUpdate(repo *model.Repo, cronJob *model.Cron) error {
+	_, err := s.engine.ID(cronJob.ID).AllCols().Update(cronJob)
+	return err
+}
+
+func (s storage) CronDelete(repo *model.Repo, id int64) error {
+	_, err := s.engine.ID(id).Where("repo_id = ?", repo.ID).Delete(new(model.Cron))
+	return err
+}
+
+// CronList return limited number of jobs based on NextExec
+// is less or equal than unitx timestamp
```

Comment:
```suggestion
```suggestion
// CronListNextExecute returns limited number of jobs with NextExec being less or equal to the provided unix timestamp
```
```